File: Online_Judge_Problems/boj/boj_1946_신입_사원/boj_1946.js
/*-----------------------------------------------------
Sub  : [BOJ] 신입 사원
Link : https://www.acmicpc.net/problem/1946
Level: 실버 1
Tag  : JS, greedy
Memo
-----------------------------------------------------*/

const TEST_MODE = true;

let input;

if (TEST_MODE) {
  const filePath = require('path').join(__dirname, 'input.txt');
  input = require('fs').readFileSync(filePath, 'utf-8').trim().split('\n');
}
else {
  input = require('fs').readFileSync(0, 'utf-8').trim().split('\n');
}

// solution
const sol = (applicants) => {
  let count = 1;
  let minInterview = applicants[0][1];

  for (let i = 1; i < applicants.length; i++) {
    if (applicants[i][1] < minInterview) {
      count++;
      minInterview = applicants[i][1];
    }
  }

  return count;
};


// input
const T = Number(input[0]);
let idx = 1;

for (let t = 0; t < T; t++) {
  let N = Number(input[idx++]);
  let applicants = [];

  for (let i = 0; i < N; i++) {
    const [doc, interview] = input[idx++].split(' ').map(Number);
    applicants.push([doc, interview]);
  }

  applicants.sort((a, b) => a[0] - b[0]);

  // solution
  const count = sol(applicants);

  // output
  console.log(count);
}

File: Online_Judge_Problems/boj/boj_1946_신입_사원/boj_1946.py
'''----------------------------------------------------
Sub  : [BOJ] 신입 사원
Link : https://www.acmicpc.net/problem/1946
Level: 실버 1
Tag  : Python, greedy
Memo
----------------------------------------------------'''

import sys

TEST_MODE = True

def get_inputs():
    if TEST_MODE:
        with open('input.txt', 'r') as f:
            for line in f:
                yield line.rstrip('\n')
    else:
        for line in sys.stdin:
            yield line.rstrip('\n')

input_gen = get_inputs()

def input():
    return next(input_gen)


# solution
def sol(score):
    score.sort(key=lambda x: x[0])    # 서류 기준 오름차순

    min_interview = score[0][1]
    count = 1

    for i in range(1, len(score)):
        if score[i][1] < min_interview:
            count += 1
            min_interview = score[i][1]

    return count


# input
T = int(input())

for _ in range(T):
    N = int(input())
    score = [list(map(int, input().split())) for _ in range(N)]

    # output
    print(sol(score))
